mozartdata/transforms/goodr_reporting/goodr_com_usage_kpis.sql

```diff
@@ -0,0 +1,198 @@
+-- noinspection SqlCurrentSchemaInspectionForFile
+
+-- noinspection SqlCurrentSchemaInspectionForFile
+
+-- noinspection SqlCurrentSchemaInspectionForFile
+
+/*
+    This table shows the daily KPIs for goodr.com usage.
+    Columns:
+        date:
+            Date reported on. This comes from dim.date to prevent issues
+            with a date not existing in Shopify.
+        sessions:
+            Total sessions as defined by Shopify for each day.
+        users:
+            Labeled "Online store visitors" in Shopify, this is the number of
+            unique users that visited the site wach day regardless of their
+            session count.
+        total_customers:
+            This is the total number of users that bought a product in a day.
+    Todos:
+        - handle leap years
+    More to come
+
+ */
+with
+    shopify_data_this_ytd as (
+                                 select
+                                     d.date::date                               as date
+                                   , shopify.sessions
+                                   , shopify."ONLINE STORE VISITORS"            as users
+                                   , shopify."NEW CUSTOMERS"                    as new_customers
+                                   , shopify.customers                          as total_customers
+                                   , shopify."SESSIONS THAT COMPLETED CHECKOUT" as sess_completed_checkout
+                                   , shopify."CONVERSION RATE"                  as daily_conversion_rate
+                                 from
+                                     dim.date                                       as d
+                                     inner join
+                                         shopify_exports.kpi_data_20240101_20250206 as shopify
+                                             on
+                                             d.date = shopify.day::date
+                                 where
+                                       d.date >= date_trunc('year', current_date)
+                                   and d.date <= current_date
```

Comment:
nothing wrong with this. Just a slightly simpler way to do it. not required your code was correct.
```suggestion
                                       date_trunc('year',d.date) = date_trunc('year', current_date)
```

Reply:
This is no longer relevant in the new file I think, since we are doing back to 2024. However, it is a smart pattern for optimizing. Noted.